python/rpdk/java/templates/init/guided_aws/StubUpdateHandler.java

```diff
@@ -29,17 +29,45 @@
         // https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/proxy/CallChain.java
 
         return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
-            // STEP 1 [first update/stabilize progress chain - required for resource update]
+
+            // STEP 1 [check if resource already exists]
+            // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
+            // if target API does not support 'ResourceNotFoundException' then following check is required
             .then(progress ->
                 // STEP 1.0 [initialize a proxy context]
+                // If your service API does not return ResourceNotFoundException on update requests against some identifier (e.g; resource Name)
+                // and instead returns a 200 even though a resource does not exist, you must first check if the resource exists here
+                // NOTE: If your service API throws 'ResourceNotFoundException' for update requests this method is not necessary
```

Comment:
Even if the service API throws `ResourceNotFoundException`, the contract tests as currently implemented expect `ResourceNotFoundException` to be thrown even when calling the handlers with obviously invalid inputs which likely will hit model validation before checking if the resource exists or not, even for services that do throw `ResourceNotFoundException`

also discussed in https://github.com/aws-cloudformation/cloudformation-cli/issues/527